src/layers/math/binary.cpp

```diff
@@ -356,8 +356,8 @@ struct greater_equal_op {
 struct and_op {
   inline DataType operator()(const DataType& x1,
                              const DataType& x2) const {
-    const bool b1 = x1 != zero && !std::isnan(x1);
-    const bool b2 = x2 != zero && !std::isnan(x2);
+    const auto& b1 = x1 >= DataType(0.5);
+    const auto& b2 = x2 >= DataType(0.5);
```

Comment:
I think that it makes more sense to keep these with the standard definition of non-zero is true and zero is false.